src/plugins/hidsim/hid_injection.cpp

```diff
@@ -204,28 +212,46 @@ static int get_display_dimensions(qmp_connection* qc, dimensions* dims)
     int matches = 0;
     int ret = 0;
 
-    /* Checks PPM-header */
-    matches = fscanf(fr, "%s", ppm_hdr);
+    /* Reads first three bytes of the potential portable pixmap */
+    matches = fscanf(fr, "%2s", ppm_hdr);
 
+    /* Checks PPM-header */
     if (matches != 1 || strcmp(ppm_hdr, PPM_MAGIC_NUMBER) != 0)
+    {
+        fprintf(stderr, "[HIDSIM]: Error reading .ppm-files's magic number\n");
+        fclose(fr);
         ret = -1;
+    }
 
-    /* Reads actual dimensions */
-    matches = fscanf(fr, "%d%d", &(dims->width), &(dims->height));
+    /* Retrieves screen dimensions by reading screen dump  */
+    uint32_t width = 0, height = 0;
+    matches = fscanf(fr, "%" SCNd32 "%" SCNd32, &width, &height);
 
-    /* Calculates pixel mapping */
-    dims->dx = calculate_pixel_unit(dims->width);
-    dims->dy = calculate_pixel_unit(dims->height);
-
-    if (matches != 2 || dims->width < 0 || dims->height < 0)
+    if (matches != 2 )
+    {
+        fprintf(stderr, "[HIDSIM]: Error reading .ppm-files metadata\n");
+        fclose(fr);
         ret = -1;
+    }
+
+    /* Sanitizes read values; no need for a screen size larger than 65k */
+    dims->width = (uint16_t) width > UINT16_MAX ? UINT16_MAX : width;
```

Comment:
Comparison result is always the same
Comparison is always false because width <= 65535.

Reply:
Comparison result is always the same
Comparison is always false because width <= 65535.

Reply:
Fixed by adding the needed brackets to cast after the ternary predicate.